src/object.js

```diff
@@ -143,6 +145,9 @@ inherits(ObjectSchema, MixedSchema, {
           return value;
         }
 
+        from = originalValue
+          ? from
+          : [...from].splice(0, 1, { schema: this, value: originalValue });
```

Comment:
`splice` returns the removed item not the array, is that what you want here?

Reply:
No, it's not what I wanted to do, thanks for the catch. I've updated this.